### DISCUSSION TITLE
Create a node-postgres adapter

### DISCUSSION THREAD

**Discussion by gue-ni** [nextauthjs/next-auth#3958] [Ideas]
### Description 📓

It would be very usefull to have a postgres adapter without having to install another package (like prism). 

### How to reproduce ☕️

Nothing here.

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by balazsorban44** [comment]
Hi, we usually require you to install an ORM, like Prisma or TypeORM, even if you use an adapter. Do I understand that you do not want to install those?

**Reply by turtlestackdev** [reply]
Not the OP, but I avoid ORMs whenever I can. Is that a hard requirement or could an adapter be created which uses a DB driver directly or a query builder such as Knex?

**Reply by balazsorban44** [reply]
Totally possible, but probably a bit harder :sweat_smile:. Here is an old attempt at a Knex adapter that might be useful: https://github.com/nextauthjs/adapters/pull/194

Otherwise, just see any of the adapters in https://github.com/nextauthjs/next-auth/tree/main/packages to get an idea

**Reply by hparra** [reply]
@turtlestackdev checkout [zapatos](https://jawj.github.io/zapatos/). I've rolled my own adapter and repos with it.

**Reply by statico** [reply]
Maybe it would be useful to include adapters specifically for Knex and Zapatos? (I'm a Knex user myself and will likely need to create my own adapter.)

**Reply by davidfaj** [reply]
Just curious, what would be some difficulties to create an adapter without an ORM (and without Knex, just using a MySQL driver, for example)? Thanks

**Comment by cymen** [comment]
If you do decide to use the Prisma-based adapter, be aware that the example/default one is not robust. See this issue for details:

https://github.com/nextauthjs/next-auth/issues/4495

I would not consider it to be production-ready. It's not hard to work around it but fair warning.
